app/controllers/analytics_controller.rb

```diff
@@ -3,21 +3,39 @@ class AnalyticsController < ApplicationController
   before_action :confirm_two_factor_authenticated
 
   def create
-    unless analytics_saved?
-      session[:platform_authenticator] = true
-      analytics.track_event(Analytics::PLATFORM_AUTHENTICATOR, results.to_h)
+    results.each do |event, result|
```

Comment:
I thought the plan was to have one event and to track all the analytics we gather under that single event.

Reply:
Or to put it another way, if eventually we gather 12 analytics data points in the javascript do you really want to log 12 analytics events?

Reply:
Probably depends on how we want to use the analytics. Separate events is the most flexible, but takes a bit more storage.

Reply:
I inferred that we wanted multiple events from the fact that the 1 event we have now is named "Platform authenticator" and not "Frontend analytics". I think that is for the best since, yeah, it lets us be more flexible.

Reply:
I just look at it from a scaling perspective.  ie 100 inserts.

Reply:
I didn't realize I left this in after I changed the format: PLATFORM_AUTHENTICATOR = 'Platform Authenticator'.freeze.  Obviously that should have been a more generic name for analytics gathering.

Reply:
I broke out the platform_authenticator into a separate key/value and I should have changed the event name

Reply:
As a side note I'm thinking we'd definitely want to log whether webauthn is supported.  Indeed that would be more important than the platform_authenticator.

Reply:
Well, what I thought we were going for here was an analytics endpoint that responded to events on the frontend and logged them. In that case, we'd only be inserting when an event is triggered on the frontend. We wouldn't be logging 100 inserts per request, but we would be logging an event for every time a user opened a dropdown or something like that.

What I was thinking for this controller was to eventually back it up by a service class that consumed the params and sorted out which events needed to be logged. Then this controller sorts through those results and logs them. I don't foresee that ever being 100s of events.

Reply:
My thoughts were different on this.  I envisioned it as a single javascript call that gathered all the relevant info much like google analytics does.  Moreover, you don't want to be making separate xhr requests for each data point.  The analytics you describe are already accounted for with the normal requests.

Reply:
We don't have a way to capture things like if a user expands a dropdown, or how many times the open the personal key confirmation modal. Where frontend analytics are concerned, those are the events that are most useful to capture.

Browser capabilities can be scooped up and made in a single request and tracked with a single event in the model I described.

Reply:
Yes.  I agree.

Reply:
The platform_authenticator falls under that umbrella (browser capability).  Can we change the event to match that?  So that we can add more (capabilities) to that single event?

Reply:
Also, I'm not quite sure why we are not using google analytics for all of this.  You can log events too.

Reply:
Renamed. PTAL